Clarify OSS definition for vendors (#4788)
open-telemetry/opentelemetry.io: content/en/ecosystem/vendors.md

customized OpenTelemetry components), that provides additional capabilities or
for improved ease of use.

> Open Source (OSS) refers to a vendor who has an observability product

Reviewer: ```suggestion
Open Source (OSS) refers to vendors who maintain an observability product
```

> that is [open source](https://opensource.org/osd). The vendor may still have
> other products that are closed source, such as a SaaS offering that
> hosts an open source product for their customers.

Reviewer: ```suggestion
is [open source](https://opensource.org/osd). The vendors may still have
other products that are closed source, such as SaaS offerings that
host open source products for their customers.
```
